src/main/python/utils/dave_reader.py

```diff
@@ -8,8 +8,10 @@
 from astropy.io import fits
```

Comment:
dave_reader.py is not very much out of control yet, but it's starting to feel like it can use some re-design.

If you have large blocks of code if(events_file) { large block } else if (light_curve_file) { large block } then perhaps composing with an interface is better.
EventsFileReader extends FileReader
LightCurveFileReader extends FileReader

if (type is events file) reader = EventsFileReader
else reader = LightCurveReader

and then you're done with all the checking, you delegate the work to the right implementation. Something like that.